internal/fswatcher/os_darwin.go

```diff
@@ -67,17 +60,19 @@ func New(root string, filter watchFilterFn, logf logFn, tomb *tomb.Tomb) (*FSWat
 	}
 
 	eventCh := make(chan Event)
-	tomb.Go(func() error {
-		for {
-			events, ok := <-es.Events
-			if !ok {
-				break
-			}
+	go func() {
+		defer close(eventCh)
+		defer close(done)
+		for events := range es.Events {
 			for i := range events {
-				event := events[i]
+				event := &events[i]
 				path := filepath.Join(mountPoint, event.Path)
-
-				if filter(path) {
+				info, err := os.Stat(path)
```

Comment:
Wouldn't this effectively skip all "Removed" event?

I don't think we can `os.Stat` the file/dir since the "removed" event will arrive when it is gone already, but we still need the remove event.

Reply:
done